spec/services/migration/migrators/application_spec.rb

```diff
@@ -219,6 +219,24 @@ def setup_failure_state
           end
         end
       end
+
+      context "when backfilling existing applications" do
+        it "sets `ecf_id`" do
+          application = create(:application, ecf_id: nil)
+
+          instance.call
+
+          expect(application.reload.ecf_id).to be_present
+        end
+
+        it "sets `cohort_id` to current cohort when schedule is not set" do
```

Comment:
could we add more tests for schedule and lead provider status?

Reply:
I think we're still missing the case where the schedule is set? (sorry if you know this/just haven't got around to it yet)

Reply:
There's a validation that prevents `application.cohort` being different from `application.schedule.cohort`
Can't actually create that scenario.
https://github.com/DFE-Digital/npq-registration/blob/main/app/models/application.rb#L208